akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
@drewhk The talk about “removing materialization” is what I consider “talking past each other”, nobody is suggesting that (and if they were then I’d provide a host of good arguments against such an endeavor). Another thing I don’t understand is the “convenience method for prematerializing stuff”, we should definitely not have such a thing. Separate materialization is only needed under rare circumstances and shall not be encouraged by the API.